cellfree/molecule/cyclomics.py

```diff
@@ -0,0 +1,187 @@
+import collections
+
+from singlecellmultiomics.molecule.featureannotatedmolecule import (
+    FeatureAnnotatedMolecule,
+)
+from singlecellmultiomics.molecule.molecule import Molecule
+
+
+class CHICMolecule(Molecule):
+    """CHIC Molecule class
+
+    Args:
+        fragments (singlecellmultiomics.fragment.Fragment): Fragments to associate to the molecule
+        **kwargs: extra args
+
+    """
+
+    def __init__(self, fragment, **kwargs):
+
+        self.site_location = None
+        Molecule.__init__(self, fragment, **kwargs)
+        self.ligation_motif = None
+
+        # Extracted from fragments:
+        self.assignment_radius = None
+
+    def _add_fragment(self, fragment):
+        # Update the cut coordinate tho the (left most extreme value)
+        self.assignment_radius = fragment.assignment_radius
+
+        if fragment.site_location is not None:
+
+            if self.site_location is None:
+                self.site_location = [
+                    fragment.site_location[0],
+                    fragment.site_location[1],
+                ]
+
+            elif fragment.strand:  # Reverse:
+                self.site_location[1] = max(
+                    fragment.site_location[1], self.site_location[1]
+                )  # this is the coordinate
+            else:
+                self.site_location[1] = min(
+                    fragment.site_location[1], self.site_location[1]
+                )  # this is the coordinate
+
+        # else : writing a fragment which has no cut location associated
+
+        Molecule._add_fragment(self, fragment)
```

Comment:
The underscore `_` may implicitly mean `do not use me outside Molecule class`. Maybe consider to make another wrapper in Molecule? For example,

In Molecule:
```python
def add_fragment(self, fragment):
    self._add_fragment(self, fragment)
```

Or simply using `self._add_fragment(self, fragment)` since `CHICMolecule` is a derivative class of `Molecule`.